pallets/inactivity-tracking/src/lib.rs

```diff
@@ -304,6 +319,60 @@ pub mod pallet {
             }
         }
 
+        /// Internal function to populate the current session active collator records with collators
+        /// part of inactive chains.
+        pub fn process_inactive_chains_for_session() {
+            match <CurrentActivityTrackingStatus<T>>::get() {
+                ActivityTrackingStatus::Disabled { .. } => return,
+                ActivityTrackingStatus::Enabled { start, end: _ } => {
+                    if start > T::CurrentSessionIndex::session_index() {
+                        return;
+                    }
+                }
+            }
+            let mut active_chains = <ActiveContainerChainsForCurrentSession<T>>::get().into_inner();
+            // Fetching the parathreads for the current session and remove them
+            // from the active chains array. In this way we handle all parathreads
+            // as inactive chains. This solution would only work if a collator either:
+            // - is assigned to one chain only
+            // - is assigned to multiple chains but all of them are parathreads
+            let parathreads_for_session = T::ParaFilter::get_parathreads_for_session();
+            active_chains.retain(|x| !parathreads_for_session.contains(x));
```

Comment:
This probably can be optimized a bit in terms of performance, since we do loop + contains for vec (another loop), it is O(m x n) time complexity. 
If we have at least 1000 container chains and at least 100 active chains it will lead to 100k iterations. Not sure if those are the realistic numbers, but I think we can convert the vec to hashset before do contains.

Reply:
Very good point! Updated `get_parathreads_for_session` to return `BTreeSet` in https://github.com/moondance-labs/tanssi/pull/899/commits/7cee1f04a04ead2e49c105029cbe691858f8e26b